state-chain/chains/src/sol/transaction_builder.rs

```diff
@@ -137,6 +148,13 @@ impl SolanaTransactionBuilder {
 
 						compute_limit += COMPUTE_UNITS_PER_FETCH_TOKEN;
 
+						let fetch_pda_and_bump =
+							derive_fetch_account(ata.address, sol_api_environment.vault_program)
+								.map_err(SolanaTransactionBuildingError::FailedToDeriveAddress)?;
+
+						// Add extra compute units for on-chain derivation
+						compute_limit += Self::derivation_compute_units(fetch_pda_and_bump.bump);
```

Comment:
think I slightly preferred the last way you had it tbh, having the function on the fetch_pda_and_bump - since it doesn't require passing anything in, and it's specific to the "Pda and bumping" and not so much to the whole builder. was there a reason you changed it? 
Not a strong opinion either way

Reply:
address it here (hopefully): https://github.com/chainflip-io/chainflip-backend/pull/5230#discussion_r1743405658

Reply:
So because this derivation shouldn't be done on PdaAndBump types directly (because it might actually be an ATA for example, we want to put it on transactionbuilder? am I understanding that correctly?

Reply:
no. Because the PdaAndBump is a primitive of Solana in a sense that it follows the Solana standard of derivation of addresses. ATA is a particular type of address derivation.
However, the compute units that we add per derivation (`derivation_compute_units`) is something we are adding as an estimation of the cost that we will incurr when deriving an address on-chain. It's acting here as a gas estimation more than anything else and it's not according to any standard/primitive of Solana.